src/rgw/rgw_file.h

```diff
@@ -502,17 +502,17 @@ namespace rgw {
 	reserve += (1 + tfh->object_name().length());
 	tfh = tfh->parent;
       }
-      bool first = true;
+      int pos = 1;
       path.reserve(reserve);
       for (auto& s : boost::adaptors::reverse(segments)) {
-	if (! first)
+	if (pos > 1) {
 	  path += "/";
-	else {
+	} else {
 	  if (!omit_bucket && (path.front() != '/')) // pretty-print
 	    path += "/";
-	  first = false;
 	}
 	path += *s;
+	++pos;
```

Comment:
i'm having trouble understanding this change from `bool first` to `int pos`; are the two not logically equivalent?

Reply:
I staged and reverted a change here, which needed to look past first position.  I decided that this logic is preferable.